### ISSUE TITLE
Android version

### ISSUE THREAD

**Issue by Rami-Sabbagh** [Sorroko/cclite#30]
Can i create an Android Version? Sorroko Accepted
For now I forked this project and I renamed it to CCLite-Android
Link : https://github.com/RamiLego4Game/CCLite-Android

Currently FileSystem Not Working, but just one way to write files (not in CC) is to use : 

file = io.open("/mnt/sdcard/etc...../filename")
file:write("Just Test".."\n") -- "\n" to create new line
file:flush()
file:close()

**Comment by lillendogge** [comment]
Omg, PLEASE keep working on this!! I have been looking for something like this for ages, and I know that many others have as well!! I understand if you no longer feel any interest in the project, but please at least make the link work! I myself do not have the time or the knowledge to be able to work on this or anything similar, sadly :/

**Comment by RamiLego4Game** [comment]
Actually i have deleted the fork respory for some reasons, but now i'm more pro at working with love2d, so i will recreate the android version :D

**Comment by Sorroko** [comment]
@lillendogge It's nice to hear there's still interest in this, I might work on this emulator a bit later

Also it may be worth just adding in Android support to the main emulator, there's not many changes that need to be made and I don't see the point of a new fork

Having said that, @RamiLego4Game feel free to fork and work on it, maybe even submit a PR

**Comment by RamiLego4Game** [comment]
CC on Android will require advanced interface with custom multitouch keyboard.
So i decided to fork Gamax version of cclite (less bugs) and convert it to a library for emulating cc, then i'll work on the android friendly interface and include the library to have the best cc emulator on android.
It will take from me hardwork and alot of time starting with 2 weeks !

**Comment by lillendogge** [comment]
Cool!! Really looking forward to it! As I said, I can't really help with coding the emulator, but tell me if there is anything that i can do to help! :)

**Comment by Sorroko** [comment]
Well after a little work the build script now creates an apk that works, I added some tweaks to get the keyboard to show and a menu bar. The screen will also scale to the size of your device whilst maintaining the aspect ratio.

The keyboard needs improvements and as mentioned above the http api doesn't work. Files are currently located at `/data/data/com.sorroko.cclite/files/save/cclite` however you will need root access, I will look into moving the save directory to the internal sdcard.

Finally a nice screenshot:
![screenshot_2015-04-06-15-16-42](https://cloud.githubusercontent.com/assets/1191498/7005637/d2c95da4-dc71-11e4-9e2d-d1fa8211bdfb.png)

I'll make a proper release soon, but in the meantime bugs & suggestions are more than welcome!

**Comment by Sorroko** [comment]
Here's a pre built apk if you would like to try it out: https://www.dropbox.com/s/mvbeveigp7kbeak/CCLite-2.2.0-alpha.apk?dl=0

**Comment by lillendogge** [comment]
Cool! I just downloaded it to try it and its awesome! I have not rooted my phone, so I can't really try To code in it, but what I did notice was that in programs like paint, the mouse either doesn't work or shows up way to the left of where you are touching the screen. Also, a suggestion for adding more advanced controls to the emulator, the top row could contain buttons for inserting Ctrl+t, Ctrl+r, etc. Or simply Ctrl. It could also contain buttons for toggeling between right/left click and showing an overlay with arrow keys. Just suggestions tho, so if you had something else in mind, then go with that first.

**Comment by Sorroko** [comment]
I have forked the love android project since it's easier to add my own changes to it, I've managed to move the save directory to /sdcard/Android/data/com.sorroko.cclite/files/save/cclite which is an improvement since you can easily access that folder. I'm still fixing a few more bugs.

As for right click, I was thinking maybe some kind of gesture? A double tap or two finger tap, some gestures might interfere with some cc programs however.

**Comment by lillendogge** [comment]
That is was i was thinking aswell, with the mouse right click. A gesture would clearly be easier to use, but would interfere with many programs. Maby go with the 'leave it to the user' option, so that you can switch between them?

**Comment by RamiLego4Game** [comment]
Note : you can use lua io library to access the sdcard, however, the limited thing that you cant get directory items :(

**Comment by RamiLego4Game** [comment]
Soory friends, my school test marks have neen send to my parents, so now no computer, no mobile to thw end of my school (5/15) (1 month) q-q.

**Comment by lillendogge** [comment]
OK, man i feel sorry for you.. :(  cya in a month i guess...

**Comment by Sorroko** [comment]
@lillendogge I agree, a double tap gesture would be perfect but it's not precise enough, I think I will just add a button for now and then add a gesture later for convenience and make it configurable

@RamiLego4Game I already have it working on the external sd card using my own fork of the library, which I also added a way to get the keyboard height finally! Ah, that's a shame, hopefully you can help when you come back.

I'll upload another test apk in a minute, feedback/bugs would be nice :)

**Comment by Sorroko** [comment]
Here you go, https://www.dropbox.com/s/5l2epo1e77vikj3/CCLite-2.2.0-beta.apk?dl=0

Files are saved in `/sdcard/Android/data/com.sorroko.cclite/files/save/cclite` as mentioned above, shortcuts such as Ctrl + S will trigger immediately since you can't hold a key down on android. I will improve the buttons soon since I can now scale and position things above the keyboard, and HTTP still needs to be fixed.

**Comment by Rami-Sabbagh** [comment]
@Sorroko Why you don't help on love android ? Alot of changes you made to it i need them ,you'll be the best.

**Comment by Sorroko** [comment]
@RamiLego4Game I might send a pull request for the keyboard height method, but I'm not sure if they want it in the main project as I think they want to keep the number of extra features to a minimum

**Comment by lillendogge** [comment]
OK, so some bugs: The shift button is off-screen and can only be seen in landscape mode (which does not look that good either), the red box that indicates that you pressed a button doesn't fit for the text, and also, did you add the key to toggle between right and left click in this version? If so, then it is not visible.

I think the biggest issue however, I's that the shift and Ctrl keys don't work as intended (as you stated previously).

PS: sorry for late answer, have been sick for a few days.
